### ISSUE TITLE
train

### ISSUE THREAD

**Issue by zhangshuaihao731** [VemundFredriksen/LungTumorMask#47]
Hello, I've been studying your LungTumorMask project recently. When I first tried training the model on the MSD Task06_Lung dataset, the final evaluation metrics were very unsatisfactory—it almost seemed like the training was unsuccessful. I noticed that the paper "Teacher-student approach for lung tumor segmentation from mixed-supervised datasets" mentions training two models, a student model and a teacher model, so why does the GitHub introduction only describe a single training process? Additionally, I'd like to ask if there have been any recent research advancements related to this LungTumorMask project? I look forward to your reply!

**Comment by VemundFredriksen** [comment]
Hi!

Thank you for taking an interest in the field and in our project!

First I'd like to emphasis that this repository originally only included the CLI tool with the pre-trained model. Since there was a lot of interest shown in the training, we decided to release the training code as is. As you might have experienced it is a big difference in code quality between the CLI tool and the training code - partially because the training code was part of our experimental phase of our master thesis and part because it was never the plan to release publicly. 

The two models are essential to the whole project. The key idea is that training a neural network on the small MSD dataset will not be sufficient, but finding huge datasets with semantic segmentations are hard - therefore we use datasets with bounding box annotations, which are cheaper, to train a teacher. The teacher can create _decent_ semantic segmentation labels because it also takes the bounding boxes as input - severely limiting the complexity of the problem. Then in turn this vast semantic segmented dataset produced by the teacher can be used to train the final model - the student.

If you took our pre-trained weights and fine tuned them on the MSD dataset and got bad results, I'm not sure why. However, if you started with a new model and tried to train it on the small MSD dataset I'm not surprised the results were poor. After all this is the entire problem the paper tries to answer.

To answer your final question; I'm sure there have been solid advancements in the field, but I'm not very up to date on it. This project have sort of fulfilled its purpose. We believe the current toolchain and model is good enough for experimental and educational use and even as a support tool to in a semi-supervised fashion. None of the authors are planning any big changes for this project in the foreseeable future.

Hope this clarified some of your questions.

**Comment by zhangshuaihao731** [comment]
Hello, I would like to know how the dataset Lung-PET-CT-Dx is filtered. Is there any code available? Or could you provide the filtered dataset?

**Comment by VemundFredriksen** [comment]
Hi @zhangshuaihao731 

I cannot remember in detail what we did, but in [this folder](https://github.com/VemundFredriksen/LungTumorMask/tree/main/training/Utils) you can see handy utils we used to format and filter files. For Lung-PET-CT-Dx this [file contains](https://github.com/VemundFredriksen/LungTumorMask/blob/main/training/Utils/PETCT_LungDx_utilities.py) som utils.

For more details you can read the method part of our master thesis which is open for download as pdf [from here](https://nva.sikt.no/registration/0198eaf6792e-e5822a04-8d2b-4438-82b1-f6121e1f3feb). Furthermore, the critereas for discarding files from the dataset is also outlined in our paper, which is also available for free download as pdf [here](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0266147).

Good luck!
